src/core/config/mle.h

```diff
@@ -261,13 +261,23 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+ * @def OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
  *
- * Define as 1 to enable Link Metrics feature.
+ * Define as 1 to enable Link Metrics initiator feature.
  *
  */
-#ifndef OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
-#define OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE 0
+#ifndef OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
+#define OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE 0
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
+ *
+ * Define as 1 to enable Link Metrics subject feature.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
+#define OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE 0
```

Comment:
Always build Link Metrics Subject for Thread 1.2 FTDs? Same way as for the CSL Transmitter.

```suggestion
#define OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE \
    OPENTRHEAD_FTD && (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
```

Reply:
@Irving-cl @abtink thoughts?

Reply:
Sounds good to me.

Just mention that:
```suggestion
#define OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE \   
    (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
```
I remember there was a discussion before that we shouldn't use `OPENTHREAD_FTD` in core configuration because core configurations are exposed to other components (for example, third party) and those components don't have the compile flags like `OPENTHREAD_FTD`. That would cause some warnings. And when we use the flag to guard the code of implementation, we use it like `OPENTHREAD_FTD && OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE`.

Reply:
@edmont enabling it conditionally under 1.2 sound fine to me.

Reply:
Fair enough. @canisLupus1313 would you please follow @Irving-cl recommendation and guard the occurrences of `OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE` with `OPENTHREAD_FTD`? Thanks.

Reply:
done.

Reply:
What I had on mind apart from the change in this file was to replace all occurances of:
`#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ISUBJECT_ENABLE`
By:
`#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MLE_LINK_METRICS_ISUBJECT_ENABLE`

You can take a look to the case of `OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE` which is very similar.

Reply:
Ok so after implementing this in proposed way with `(OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)` unfortunately this change introduces [ot-nrf528xx](https://github.com/openthread/ot-nrf528xx) repo with linking issues of final example binaries.

The thing is that ot-nrf repo do not generate different object files for FTD, MTD, RCP like openthread one does. Up until now OPENTHREAD_FTD (or similar) wasn't used in this platform repository so the problem didn't occur.

To fix this issue I will have to introduce significant changes in build process to generate different object files for different types of devices (FTD, MTD etc.). However I prefer to create a separate PR for this.

Reply:
Fair enough, this change can be applied later.